### PR TITLE
feature/PATRO23-206 - add DateTimeProvider.

### PR DESCRIPTION
**Pull request by IzabelaPalubicka**
https://tracker.intive.com/jira/browse/PATRO23-206

### CODE REVIEW

File: src/shared/abstractions/DateTimeProvider.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+/// </summary>
+public class DateTimeProvider : IDateTimeProvider
+{
+	private TimeZoneInfo timeZone = TimeZoneInfo.Local;
+
+	/// <inheritdoc/>
+	public DateTime UtcNow() => DateTime.UtcNow;
+
+	/// <inheritdoc/>
+	public DateTime LocalNow()
```
Comment (mateuszgiedzinski) [inline src/shared/abstractions/DateTimeProvider.cs:14]: Methods are fine but in C# we can specify it as getter.

File: src/shared/abstractions/IDateTimeProvider.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+/// <summary>
+/// Interface that defines methods related to date and time.
+/// </summary>
+public interface IDateTimeProvider
+{
+	/// <summary>
+	/// Get UTC date and time.
+	/// </summary>
+	/// <returns>Utc date and time.</returns>
+	DateTime UtcNow();
```
Comment (mateuszgiedzinski) [inline src/shared/abstractions/IDateTimeProvider.cs:12]: DateTime UtcNow {get;} maybe?

File: tests/unit/Intive.Patronage2023.Example.Application.Tests/DateTimeProviderTests.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+public class DateTimeProviderTests
+{
+	/// <summary>
+	/// Test that check if the method "LocalNow" returns the correct date and time when UtcNow has fixed value.
+	/// </summary>
+	[Fact]
+	public void LocalNow_WhenUtcNowHasFixedValue_ShouldReturnLocalTime()
+	{
+		// Arrange
+		var dateTimeProvider = new Mock<IDateTimeProvider>();
```
Comment (mateuszgiedzinski) [inline tests/unit/Intive.Patronage2023.Example.Application.Tests/DateTimeProviderTests.cs:19]: Test must be rewritten. 
First of all in test we don't want mock tested provider. Setup tested instance in test class ctor. 
If you need to test if settimezone works fine you can first get utc then get local and then convert utc to local by your own as you did. And in comparison you can use fluent assertion and this method 
.Should().BeCloseTo()

File: src/shared/abstractions/DateTimeProvider.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+namespace Intive.Patronage2023.Shared.Abstractions;
+
+/// <summary>
+/// Implementation of the IDateTimeService interface.
+/// </summary>
+public class DateTimeProvider : IDateTimeProvider
```
Comment (mateuszgiedzinski) [inline src/shared/abstractions/DateTimeProvider.cs:6]: hmm as far as I know implementation shouldn't be part of abstraction project.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/shared/infrastructure/SharedModule.cs [hunk 5]
```diff
@@ -21,6 +22,9 @@ public static IServiceCollection AddSharedModule(this IServiceCollection service
 		services.AddTransient(typeof(INotificationHandler<>), typeof(MediatrEventHandlerAdapter<>));
 		services.AddTransient(typeof(IRequestHandler<,>), typeof(MediatRQueryHandlerAdapter<,>));
 		services.AddTransient(typeof(IRequestHandler<>), typeof(MediatRCommandHandlerAdapter<>));
+
+		services.AddSingleton<IDateTimeProvider, DateTimeProvider>();
```
Comment (AlanP22) [inline src/shared/infrastructure/SharedModule.cs:26]: Shouldn't it be better to add it as scoped?

Comment (IzabelaPalubicka) [reply src/shared/infrastructure/SharedModule.cs:26]: Why do you think that?

Comment (mateuszgiedzinski) [reply src/shared/infrastructure/SharedModule.cs:26]: For now it could be singleton. In the future it should be Scoped to set the timezone.

**Review by mateuszgiedzinski — APPROVED** [review]

**Pull request merged**